# **Importing Data Frame**

In [1]:
import numpy as np
import pandas as pd


In [2]:
from google.colab import files
uploaded = files.upload()

Saving movies.csv to movies.csv


In [3]:
import io
df2 = pd.read_csv(io.BytesIO(uploaded['movies.csv']))
# Dataset is now stored in a Pandas Dataframe
df2

MOVIES         YEAR  \
0                           Blood Red Sky       (2021)   
1     Masters of the Universe: Revelation     (2021– )   
2                        The Walking Dead  (2010–2022)   
3                          Rick and Morty     (2013– )   
4                         Army of Thieves       (2021)   
...                                   ...          ...   
9994                       The Imperfects     (2021– )   
9995                               Arcane     (2021– )   
9996                    Heart of Invictus     (2022– )   
9997                       The Imperfects     (2021– )   
9998                       The Imperfects     (2021– )   

                                           GENRE  RATING  \
0         \nAction, Horror, Thriller                 6.1   
1     \nAnimation, Action, Adventure                 5.0   
2          \nDrama, Horror, Thriller                 8.2   
3     \nAnimation, Adventure, Comedy                 9.2   
4            \nAction, Crime, Horror                 NaN   
...                                          ...     ...   
9994     \nAdventure, Drama, Fantasy                 NaN   
9995  \nAnimation, Action, Adventure                 NaN   
9996            \nDocumentary, Sport                 NaN   
9997     \nAdventure, Drama, Fantasy                 NaN   
9998     \nAdventure, Drama, Fantasy                 NaN   

                                               ONE-LINE  \
0     \nA woman with a mysterious illness is forced ...   
1     \nThe war for Eternia begins again in what may...   
2     \nSheriff Deputy Rick Grimes wakes up from a c...   
3     \nAn animated series that follows the exploits...   
4     \nA prequel, set before the events of Army of ...   
...                                                 ...   
9994                                     \nAdd a Plot\n   
9995                                     \nAdd a Plot\n   
9996                                     \nAdd a Plot\n   
9997                                     \nAdd a Plot\n   
9998                                     \nAdd a Plot\n   

                                                  STARS    VOTES  RunTime  \
0     \n    Director:\nPeter Thorwarth\n| \n    Star...   21,062    121.0   
1     \n            \n    Stars:\nChris Wood, \nSara...   17,870     25.0   
2     \n            \n    Stars:\nAndrew Lincoln, \n...  885,805     44.0   
3     \n            \n    Stars:\nJustin Roiland, \n...  414,849     23.0   
4     \n    Director:\nMatthias Schweighöfer\n| \n  ...      NaN      NaN   
...                                                 ...      ...      ...   
9994  \n            \n    Stars:\nMorgan Taylor Camp...      NaN      NaN   
9995                                                 \n      NaN      NaN   
9996  \n    Director:\nOrlando von Einsiedel\n| \n  ...      NaN      NaN   
9997  \n    Director:\nJovanka Vuckovic\n| \n    Sta...      NaN      NaN   
9998  \n    Director:\nJovanka Vuckovic\n| \n    Sta...      NaN      NaN   

     Gross  
0      NaN  
1      NaN  
2      NaN  
3      NaN  
4      NaN  
...    ...  
9994   NaN  
9995   NaN  
9996   NaN  
9997   NaN  
9998   NaN  

[9999 rows x 9 columns]

# **Taking a look at the data**

In [4]:
df2.head()

MOVIES         YEAR  \
0                        Blood Red Sky       (2021)   
1  Masters of the Universe: Revelation     (2021– )   
2                     The Walking Dead  (2010–2022)   
3                       Rick and Morty     (2013– )   
4                      Army of Thieves       (2021)   

                                        GENRE  RATING  \
0      \nAction, Horror, Thriller                 6.1   
1  \nAnimation, Action, Adventure                 5.0   
2       \nDrama, Horror, Thriller                 8.2   
3  \nAnimation, Adventure, Comedy                 9.2   
4         \nAction, Crime, Horror                 NaN   

                                            ONE-LINE  \
0  \nA woman with a mysterious illness is forced ...   
1  \nThe war for Eternia begins again in what may...   
2  \nSheriff Deputy Rick Grimes wakes up from a c...   
3  \nAn animated series that follows the exploits...   
4  \nA prequel, set before the events of Army of ...   

                                               STARS    VOTES  RunTime Gross  
0  \n    Director:\nPeter Thorwarth\n| \n    Star...   21,062    121.0   NaN  
1  \n            \n    Stars:\nChris Wood, \nSara...   17,870     25.0   NaN  
2  \n            \n    Stars:\nAndrew Lincoln, \n...  885,805     44.0   NaN  
3  \n            \n    Stars:\nJustin Roiland, \n...  414,849     23.0   NaN  
4  \n    Director:\nMatthias Schweighöfer\n| \n  ...      NaN      NaN   NaN

Initial notes:

 

*   Year dates aren't in the same format. Some have only one year. Will need to remove the brackets.
*   Genre and stars have/ start with /n which needs to be removed and there are 3 or more objects in one cell (list) which may need to be seperated for better analysis


*   Rating, Votes and Runtime have some NaN's. can check if its necessary to remove these/ fill these
*   One- line has /n which needs to be removed. need to consider if this column is neccessarry

* Gross seems to have multiple NaN's. Will need to check the percentage to see if this column can be kept










In [5]:
df2.describe(include = 'object')

MOVIES      YEAR                 GENRE        ONE-LINE  \
count               9999      9355                  9919            9999   
unique              6817       438                   510            8688   
top      Bleach: Burîchi  (2020– )  \nComedy              \nAdd a Plot\n   
freq                  65       892                   852            1265   

       STARS VOTES   Gross  
count   9999  8179     460  
unique  7877  4129     332  
top       \n     7  $0.01M  
freq     456    35      22

Thoughts:



* Gross has 460 count (out of 9999) which suggests that majority of the data is null. Will need to check for nulls to clarify this.
*   /n needs to be removed from multiple columns.



In [6]:
df2.dtypes

MOVIES       object
YEAR         object
GENRE        object
RATING      float64
ONE-LINE     object
STARS        object
VOTES        object
RunTime     float64
Gross        object
dtype: object

Thoughts:

Year, Votes and Gross are the wrong data types 

In [7]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9999 entries, 0 to 9998
Data columns (total 9 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   MOVIES    9999 non-null   object 
 1   YEAR      9355 non-null   object 
 2   GENRE     9919 non-null   object 
 3   RATING    8179 non-null   float64
 4   ONE-LINE  9999 non-null   object 
 5   STARS     9999 non-null   object 
 6   VOTES     8179 non-null   object 
 7   RunTime   7041 non-null   float64
 8   Gross     460 non-null    object 
dtypes: float64(2), object(7)
memory usage: 703.2+ KB


In [8]:
df2.isnull().sum()

MOVIES         0
YEAR         644
GENRE         80
RATING      1820
ONE-LINE       0
STARS          0
VOTES       1820
RunTime     2958
Gross       9539
dtype: int64

Thoughts:


*   Gross has too many non- null values and so needs to be removed


*   Dont need to remove other columns unless deemed unneccessary to have them as null values not large enough to affect data





**To do:**



*   Remove Gross column due to too many null values. will remove one-line as i believe it is unneccessary for analysis

*   Remove brackets from Year column

*   Index year values to keep only first year, Labeled Start Year. The ones that have multiple years are correct - due to being series movies - but can add a column to say series or not. Yes or no. (if neccessary)

*   Remove /n from genre, stars and one-line

*   Split Genre and Stars into multiple more lines/ cells so that it is one word/ genre/ star per cell
*   Need to decide to backfill or frontfill rating, runtime and votes NaN


*   Change data type for year and votes


In [9]:
df2.columns

Index(['MOVIES', 'YEAR', 'GENRE', 'RATING', 'ONE-LINE', 'STARS', 'VOTES',
       'RunTime', 'Gross'],
      dtype='object')

# **Cleaning the Data**

## Remove Gross and One-line column


In [10]:
df2.drop(columns = ['Gross', 'ONE-LINE'], axis = 'columns', inplace = True)

In [11]:
df2

MOVIES         YEAR  \
0                           Blood Red Sky       (2021)   
1     Masters of the Universe: Revelation     (2021– )   
2                        The Walking Dead  (2010–2022)   
3                          Rick and Morty     (2013– )   
4                         Army of Thieves       (2021)   
...                                   ...          ...   
9994                       The Imperfects     (2021– )   
9995                               Arcane     (2021– )   
9996                    Heart of Invictus     (2022– )   
9997                       The Imperfects     (2021– )   
9998                       The Imperfects     (2021– )   

                                           GENRE  RATING  \
0         \nAction, Horror, Thriller                 6.1   
1     \nAnimation, Action, Adventure                 5.0   
2          \nDrama, Horror, Thriller                 8.2   
3     \nAnimation, Adventure, Comedy                 9.2   
4            \nAction, Crime, Horror                 NaN   
...                                          ...     ...   
9994     \nAdventure, Drama, Fantasy                 NaN   
9995  \nAnimation, Action, Adventure                 NaN   
9996            \nDocumentary, Sport                 NaN   
9997     \nAdventure, Drama, Fantasy                 NaN   
9998     \nAdventure, Drama, Fantasy                 NaN   

                                                  STARS    VOTES  RunTime  
0     \n    Director:\nPeter Thorwarth\n| \n    Star...   21,062    121.0  
1     \n            \n    Stars:\nChris Wood, \nSara...   17,870     25.0  
2     \n            \n    Stars:\nAndrew Lincoln, \n...  885,805     44.0  
3     \n            \n    Stars:\nJustin Roiland, \n...  414,849     23.0  
4     \n    Director:\nMatthias Schweighöfer\n| \n  ...      NaN      NaN  
...                                                 ...      ...      ...  
9994  \n            \n    Stars:\nMorgan Taylor Camp...      NaN      NaN  
9995                                                 \n      NaN      NaN  
9996  \n    Director:\nOrlando von Einsiedel\n| \n  ...      NaN      NaN  
9997  \n    Director:\nJovanka Vuckovic\n| \n    Sta...      NaN      NaN  
9998  \n    Director:\nJovanka Vuckovic\n| \n    Sta...      NaN      NaN  

[9999 rows x 7 columns]

## Remove brackets from Year column

In [12]:
df2.YEAR

0            (2021)
1          (2021– )
2       (2010–2022)
3          (2013– )
4            (2021)
           ...     
9994       (2021– )
9995       (2021– )
9996       (2022– )
9997       (2021– )
9998       (2021– )
Name: YEAR, Length: 9999, dtype: object

In [13]:
df2['YEAR'] = df2['YEAR'].str.strip('(').str.strip(')')
df2['YEAR']

0            2021
1          2021– 
2       2010–2022
3          2013– 
4            2021
          ...    
9994       2021– 
9995       2021– 
9996       2022– 
9997       2021– 
9998       2021– 
Name: YEAR, Length: 9999, dtype: object

## Index year values to keep only first year, Labeled Start Year.

In [14]:
df2['Start Year'] = df2['YEAR'].str[:4]
df2['Start Year']

0       2021
1       2021
2       2010
3       2013
4       2021
        ... 
9994    2021
9995    2021
9996    2022
9997    2021
9998    2021
Name: Start Year, Length: 9999, dtype: object

In [15]:
df2

MOVIES       YEAR  \
0                           Blood Red Sky       2021   
1     Masters of the Universe: Revelation     2021–    
2                        The Walking Dead  2010–2022   
3                          Rick and Morty     2013–    
4                         Army of Thieves       2021   
...                                   ...        ...   
9994                       The Imperfects     2021–    
9995                               Arcane     2021–    
9996                    Heart of Invictus     2022–    
9997                       The Imperfects     2021–    
9998                       The Imperfects     2021–    

                                           GENRE  RATING  \
0         \nAction, Horror, Thriller                 6.1   
1     \nAnimation, Action, Adventure                 5.0   
2          \nDrama, Horror, Thriller                 8.2   
3     \nAnimation, Adventure, Comedy                 9.2   
4            \nAction, Crime, Horror                 NaN   
...                                          ...     ...   
9994     \nAdventure, Drama, Fantasy                 NaN   
9995  \nAnimation, Action, Adventure                 NaN   
9996            \nDocumentary, Sport                 NaN   
9997     \nAdventure, Drama, Fantasy                 NaN   
9998     \nAdventure, Drama, Fantasy                 NaN   

                                                  STARS    VOTES  RunTime  \
0     \n    Director:\nPeter Thorwarth\n| \n    Star...   21,062    121.0   
1     \n            \n    Stars:\nChris Wood, \nSara...   17,870     25.0   
2     \n            \n    Stars:\nAndrew Lincoln, \n...  885,805     44.0   
3     \n            \n    Stars:\nJustin Roiland, \n...  414,849     23.0   
4     \n    Director:\nMatthias Schweighöfer\n| \n  ...      NaN      NaN   
...                                                 ...      ...      ...   
9994  \n            \n    Stars:\nMorgan Taylor Camp...      NaN      NaN   
9995                                                 \n      NaN      NaN   
9996  \n    Director:\nOrlando von Einsiedel\n| \n  ...      NaN      NaN   
9997  \n    Director:\nJovanka Vuckovic\n| \n    Sta...      NaN      NaN   
9998  \n    Director:\nJovanka Vuckovic\n| \n    Sta...      NaN      NaN   

     Start Year  
0          2021  
1          2021  
2          2010  
3          2013  
4          2021  
...         ...  
9994       2021  
9995       2021  
9996       2022  
9997       2021  
9998       2021  

[9999 rows x 8 columns]

## Create a 'series' column based on Year column

In [16]:
df2['char_length'] = df2['YEAR'].str.len()
df2

MOVIES       YEAR  \
0                           Blood Red Sky       2021   
1     Masters of the Universe: Revelation     2021–    
2                        The Walking Dead  2010–2022   
3                          Rick and Morty     2013–    
4                         Army of Thieves       2021   
...                                   ...        ...   
9994                       The Imperfects     2021–    
9995                               Arcane     2021–    
9996                    Heart of Invictus     2022–    
9997                       The Imperfects     2021–    
9998                       The Imperfects     2021–    

                                           GENRE  RATING  \
0         \nAction, Horror, Thriller                 6.1   
1     \nAnimation, Action, Adventure                 5.0   
2          \nDrama, Horror, Thriller                 8.2   
3     \nAnimation, Adventure, Comedy                 9.2   
4            \nAction, Crime, Horror                 NaN   
...                                          ...     ...   
9994     \nAdventure, Drama, Fantasy                 NaN   
9995  \nAnimation, Action, Adventure                 NaN   
9996            \nDocumentary, Sport                 NaN   
9997     \nAdventure, Drama, Fantasy                 NaN   
9998     \nAdventure, Drama, Fantasy                 NaN   

                                                  STARS    VOTES  RunTime  \
0     \n    Director:\nPeter Thorwarth\n| \n    Star...   21,062    121.0   
1     \n            \n    Stars:\nChris Wood, \nSara...   17,870     25.0   
2     \n            \n    Stars:\nAndrew Lincoln, \n...  885,805     44.0   
3     \n            \n    Stars:\nJustin Roiland, \n...  414,849     23.0   
4     \n    Director:\nMatthias Schweighöfer\n| \n  ...      NaN      NaN   
...                                                 ...      ...      ...   
9994  \n            \n    Stars:\nMorgan Taylor Camp...      NaN      NaN   
9995                                                 \n      NaN      NaN   
9996  \n    Director:\nOrlando von Einsiedel\n| \n  ...      NaN      NaN   
9997  \n    Director:\nJovanka Vuckovic\n| \n    Sta...      NaN      NaN   
9998  \n    Director:\nJovanka Vuckovic\n| \n    Sta...      NaN      NaN   

     Start Year  char_length  
0          2021          4.0  
1          2021          6.0  
2          2010          9.0  
3          2013          6.0  
4          2021          4.0  
...         ...          ...  
9994       2021          6.0  
9995       2021          6.0  
9996       2022          6.0  
9997       2021          6.0  
9998       2021          6.0  

[9999 rows x 9 columns]

In [17]:
df2['Series'] = np.where(df2['char_length'] > 4, True, False)
df2.head()

MOVIES       YEAR  \
0                        Blood Red Sky       2021   
1  Masters of the Universe: Revelation     2021–    
2                     The Walking Dead  2010–2022   
3                       Rick and Morty     2013–    
4                      Army of Thieves       2021   

                                        GENRE  RATING  \
0      \nAction, Horror, Thriller                 6.1   
1  \nAnimation, Action, Adventure                 5.0   
2       \nDrama, Horror, Thriller                 8.2   
3  \nAnimation, Adventure, Comedy                 9.2   
4         \nAction, Crime, Horror                 NaN   

                                               STARS    VOTES  RunTime  \
0  \n    Director:\nPeter Thorwarth\n| \n    Star...   21,062    121.0   
1  \n            \n    Stars:\nChris Wood, \nSara...   17,870     25.0   
2  \n            \n    Stars:\nAndrew Lincoln, \n...  885,805     44.0   
3  \n            \n    Stars:\nJustin Roiland, \n...  414,849     23.0   
4  \n    Director:\nMatthias Schweighöfer\n| \n  ...      NaN      NaN   

  Start Year  char_length  Series  
0       2021          4.0   False  
1       2021          6.0    True  
2       2010          9.0    True  
3       2013          6.0    True  
4       2021          4.0   False

In [18]:
#Will remove 'year' and 'char_length' columns as they are unneeded but will create a new data frame to continue with this
df3 = df2.drop(columns = ['YEAR','char_length'], axis = 'columns',)
df3

MOVIES  \
0                           Blood Red Sky   
1     Masters of the Universe: Revelation   
2                        The Walking Dead   
3                          Rick and Morty   
4                         Army of Thieves   
...                                   ...   
9994                       The Imperfects   
9995                               Arcane   
9996                    Heart of Invictus   
9997                       The Imperfects   
9998                       The Imperfects   

                                           GENRE  RATING  \
0         \nAction, Horror, Thriller                 6.1   
1     \nAnimation, Action, Adventure                 5.0   
2          \nDrama, Horror, Thriller                 8.2   
3     \nAnimation, Adventure, Comedy                 9.2   
4            \nAction, Crime, Horror                 NaN   
...                                          ...     ...   
9994     \nAdventure, Drama, Fantasy                 NaN   
9995  \nAnimation, Action, Adventure                 NaN   
9996            \nDocumentary, Sport                 NaN   
9997     \nAdventure, Drama, Fantasy                 NaN   
9998     \nAdventure, Drama, Fantasy                 NaN   

                                                  STARS    VOTES  RunTime  \
0     \n    Director:\nPeter Thorwarth\n| \n    Star...   21,062    121.0   
1     \n            \n    Stars:\nChris Wood, \nSara...   17,870     25.0   
2     \n            \n    Stars:\nAndrew Lincoln, \n...  885,805     44.0   
3     \n            \n    Stars:\nJustin Roiland, \n...  414,849     23.0   
4     \n    Director:\nMatthias Schweighöfer\n| \n  ...      NaN      NaN   
...                                                 ...      ...      ...   
9994  \n            \n    Stars:\nMorgan Taylor Camp...      NaN      NaN   
9995                                                 \n      NaN      NaN   
9996  \n    Director:\nOrlando von Einsiedel\n| \n  ...      NaN      NaN   
9997  \n    Director:\nJovanka Vuckovic\n| \n    Sta...      NaN      NaN   
9998  \n    Director:\nJovanka Vuckovic\n| \n    Sta...      NaN      NaN   

     Start Year  Series  
0          2021   False  
1          2021    True  
2          2010    True  
3          2013    True  
4          2021   False  
...         ...     ...  
9994       2021    True  
9995       2021    True  
9996       2022    True  
9997       2021    True  
9998       2021    True  

[9999 rows x 8 columns]

In [19]:
#Will change True/ false to yes/no
df3 =df3.replace({'Series': {True : 'yes', False: 'no'}})
df3

MOVIES  \
0                           Blood Red Sky   
1     Masters of the Universe: Revelation   
2                        The Walking Dead   
3                          Rick and Morty   
4                         Army of Thieves   
...                                   ...   
9994                       The Imperfects   
9995                               Arcane   
9996                    Heart of Invictus   
9997                       The Imperfects   
9998                       The Imperfects   

                                           GENRE  RATING  \
0         \nAction, Horror, Thriller                 6.1   
1     \nAnimation, Action, Adventure                 5.0   
2          \nDrama, Horror, Thriller                 8.2   
3     \nAnimation, Adventure, Comedy                 9.2   
4            \nAction, Crime, Horror                 NaN   
...                                          ...     ...   
9994     \nAdventure, Drama, Fantasy                 NaN   
9995  \nAnimation, Action, Adventure                 NaN   
9996            \nDocumentary, Sport                 NaN   
9997     \nAdventure, Drama, Fantasy                 NaN   
9998     \nAdventure, Drama, Fantasy                 NaN   

                                                  STARS    VOTES  RunTime  \
0     \n    Director:\nPeter Thorwarth\n| \n    Star...   21,062    121.0   
1     \n            \n    Stars:\nChris Wood, \nSara...   17,870     25.0   
2     \n            \n    Stars:\nAndrew Lincoln, \n...  885,805     44.0   
3     \n            \n    Stars:\nJustin Roiland, \n...  414,849     23.0   
4     \n    Director:\nMatthias Schweighöfer\n| \n  ...      NaN      NaN   
...                                                 ...      ...      ...   
9994  \n            \n    Stars:\nMorgan Taylor Camp...      NaN      NaN   
9995                                                 \n      NaN      NaN   
9996  \n    Director:\nOrlando von Einsiedel\n| \n  ...      NaN      NaN   
9997  \n    Director:\nJovanka Vuckovic\n| \n    Sta...      NaN      NaN   
9998  \n    Director:\nJovanka Vuckovic\n| \n    Sta...      NaN      NaN   

     Start Year Series  
0          2021     no  
1          2021    yes  
2          2010    yes  
3          2013    yes  
4          2021     no  
...         ...    ...  
9994       2021    yes  
9995       2021    yes  
9996       2022    yes  
9997       2021    yes  
9998       2021    yes  

[9999 rows x 8 columns]

## Remove /n from Genre

In [20]:
df3.columns

Index(['MOVIES', 'GENRE', 'RATING', 'STARS', 'VOTES', 'RunTime', 'Start Year',
       'Series'],
      dtype='object')

In [21]:
df3.GENRE

0           \nAction, Horror, Thriller            
1       \nAnimation, Action, Adventure            
2            \nDrama, Horror, Thriller            
3       \nAnimation, Adventure, Comedy            
4              \nAction, Crime, Horror            
                           ...                    
9994       \nAdventure, Drama, Fantasy            
9995    \nAnimation, Action, Adventure            
9996              \nDocumentary, Sport            
9997       \nAdventure, Drama, Fantasy            
9998       \nAdventure, Drama, Fantasy            
Name: GENRE, Length: 9999, dtype: object

In [22]:
df3['GENRE'] = df3['GENRE'].str.strip('\n')
df3

MOVIES  \
0                           Blood Red Sky   
1     Masters of the Universe: Revelation   
2                        The Walking Dead   
3                          Rick and Morty   
4                         Army of Thieves   
...                                   ...   
9994                       The Imperfects   
9995                               Arcane   
9996                    Heart of Invictus   
9997                       The Imperfects   
9998                       The Imperfects   

                                         GENRE  RATING  \
0         Action, Horror, Thriller                 6.1   
1     Animation, Action, Adventure                 5.0   
2          Drama, Horror, Thriller                 8.2   
3     Animation, Adventure, Comedy                 9.2   
4            Action, Crime, Horror                 NaN   
...                                        ...     ...   
9994     Adventure, Drama, Fantasy                 NaN   
9995  Animation, Action, Adventure                 NaN   
9996            Documentary, Sport                 NaN   
9997     Adventure, Drama, Fantasy                 NaN   
9998     Adventure, Drama, Fantasy                 NaN   

                                                  STARS    VOTES  RunTime  \
0     \n    Director:\nPeter Thorwarth\n| \n    Star...   21,062    121.0   
1     \n            \n    Stars:\nChris Wood, \nSara...   17,870     25.0   
2     \n            \n    Stars:\nAndrew Lincoln, \n...  885,805     44.0   
3     \n            \n    Stars:\nJustin Roiland, \n...  414,849     23.0   
4     \n    Director:\nMatthias Schweighöfer\n| \n  ...      NaN      NaN   
...                                                 ...      ...      ...   
9994  \n            \n    Stars:\nMorgan Taylor Camp...      NaN      NaN   
9995                                                 \n      NaN      NaN   
9996  \n    Director:\nOrlando von Einsiedel\n| \n  ...      NaN      NaN   
9997  \n    Director:\nJovanka Vuckovic\n| \n    Sta...      NaN      NaN   
9998  \n    Director:\nJovanka Vuckovic\n| \n    Sta...      NaN      NaN   

     Start Year Series  
0          2021     no  
1          2021    yes  
2          2010    yes  
3          2013    yes  
4          2021     no  
...         ...    ...  
9994       2021    yes  
9995       2021    yes  
9996       2022    yes  
9997       2021    yes  
9998       2021    yes  

[9999 rows x 8 columns]

## Adding delimeters so that Stars can be split into into Director and Actors



In [26]:
df3

MOVIES  \
0                           Blood Red Sky   
1     Masters of the Universe: Revelation   
2                        The Walking Dead   
3                          Rick and Morty   
4                         Army of Thieves   
...                                   ...   
9994                       The Imperfects   
9995                               Arcane   
9996                    Heart of Invictus   
9997                       The Imperfects   
9998                       The Imperfects   

                                         GENRE  RATING  \
0         Action, Horror, Thriller                 6.1   
1     Animation, Action, Adventure                 5.0   
2          Drama, Horror, Thriller                 8.2   
3     Animation, Adventure, Comedy                 9.2   
4            Action, Crime, Horror                 NaN   
...                                        ...     ...   
9994     Adventure, Drama, Fantasy                 NaN   
9995  Animation, Action, Adventure                 NaN   
9996            Documentary, Sport                 NaN   
9997     Adventure, Drama, Fantasy                 NaN   
9998     Adventure, Drama, Fantasy                 NaN   

                                                  STARS    VOTES  RunTime  \
0     \n    Director:\nPeter Thorwarth\n| \n    Star...   21,062    121.0   
1     \n            \n    Stars:\nChris Wood, \nSara...   17,870     25.0   
2     \n            \n    Stars:\nAndrew Lincoln, \n...  885,805     44.0   
3     \n            \n    Stars:\nJustin Roiland, \n...  414,849     23.0   
4     \n    Director:\nMatthias Schweighöfer\n| \n  ...      NaN      NaN   
...                                                 ...      ...      ...   
9994  \n            \n    Stars:\nMorgan Taylor Camp...      NaN      NaN   
9995                                                 \n      NaN      NaN   
9996  \n    Director:\nOrlando von Einsiedel\n| \n  ...      NaN      NaN   
9997  \n    Director:\nJovanka Vuckovic\n| \n    Sta...      NaN      NaN   
9998  \n    Director:\nJovanka Vuckovic\n| \n    Sta...      NaN      NaN   

     Start Year Series  
0          2021     no  
1          2021    yes  
2          2010    yes  
3          2013    yes  
4          2021     no  
...         ...    ...  
9994       2021    yes  
9995       2021    yes  
9996       2022    yes  
9997       2021    yes  
9998       2021    yes  

[9999 rows x 8 columns]

In [28]:
df3['STARS'] = df3['STARS'].str.replace('Director:\n','0 ', regex=False).str.replace('Stars:\n','1 ', regex=False)
df3

MOVIES  \
0                           Blood Red Sky   
1     Masters of the Universe: Revelation   
2                        The Walking Dead   
3                          Rick and Morty   
4                         Army of Thieves   
...                                   ...   
9994                       The Imperfects   
9995                               Arcane   
9996                    Heart of Invictus   
9997                       The Imperfects   
9998                       The Imperfects   

                                         GENRE  RATING  \
0         Action, Horror, Thriller                 6.1   
1     Animation, Action, Adventure                 5.0   
2          Drama, Horror, Thriller                 8.2   
3     Animation, Adventure, Comedy                 9.2   
4            Action, Crime, Horror                 NaN   
...                                        ...     ...   
9994     Adventure, Drama, Fantasy                 NaN   
9995  Animation, Action, Adventure                 NaN   
9996            Documentary, Sport                 NaN   
9997     Adventure, Drama, Fantasy                 NaN   
9998     Adventure, Drama, Fantasy                 NaN   

                                                  STARS    VOTES  RunTime  \
0     \n    0 Peter Thorwarth\n| \n    1 Peri Baumei...   21,062    121.0   
1     \n            \n    1 Chris Wood, \nSarah Mich...   17,870     25.0   
2     \n            \n    1 Andrew Lincoln, \nNorman...  885,805     44.0   
3     \n            \n    1 Justin Roiland, \nChris ...  414,849     23.0   
4     \n    0 Matthias Schweighöfer\n| \n    1 Matth...      NaN      NaN   
...                                                 ...      ...      ...   
9994  \n            \n    1 Morgan Taylor Campbell, ...      NaN      NaN   
9995                                                 \n      NaN      NaN   
9996  \n    0 Orlando von Einsiedel\n| \n    Star:\n...      NaN      NaN   
9997  \n    0 Jovanka Vuckovic\n| \n    1 Morgan Tay...      NaN      NaN   
9998  \n    0 Jovanka Vuckovic\n| \n    1 Morgan Tay...      NaN      NaN   

     Start Year Series  
0          2021     no  
1          2021    yes  
2          2010    yes  
3          2013    yes  
4          2021     no  
...         ...    ...  
9994       2021    yes  
9995       2021    yes  
9996       2022    yes  
9997       2021    yes  
9998       2021    yes  

[9999 rows x 8 columns]

In [29]:
df3['STARS'] = df3['STARS'].str.replace('\n','N/A', regex=False)
df3

MOVIES  \
0                           Blood Red Sky   
1     Masters of the Universe: Revelation   
2                        The Walking Dead   
3                          Rick and Morty   
4                         Army of Thieves   
...                                   ...   
9994                       The Imperfects   
9995                               Arcane   
9996                    Heart of Invictus   
9997                       The Imperfects   
9998                       The Imperfects   

                                         GENRE  RATING  \
0         Action, Horror, Thriller                 6.1   
1     Animation, Action, Adventure                 5.0   
2          Drama, Horror, Thriller                 8.2   
3     Animation, Adventure, Comedy                 9.2   
4            Action, Crime, Horror                 NaN   
...                                        ...     ...   
9994     Adventure, Drama, Fantasy                 NaN   
9995  Animation, Action, Adventure                 NaN   
9996            Documentary, Sport                 NaN   
9997     Adventure, Drama, Fantasy                 NaN   
9998     Adventure, Drama, Fantasy                 NaN   

                                                  STARS    VOTES  RunTime  \
0     N/A    0 Peter ThorwarthN/A| N/A    1 Peri Bau...   21,062    121.0   
1     N/A            N/A    1 Chris Wood, N/ASarah M...   17,870     25.0   
2     N/A            N/A    1 Andrew Lincoln, N/ANor...  885,805     44.0   
3     N/A            N/A    1 Justin Roiland, N/AChr...  414,849     23.0   
4     N/A    0 Matthias SchweighöferN/A| N/A    1 Ma...      NaN      NaN   
...                                                 ...      ...      ...   
9994  N/A            N/A    1 Morgan Taylor Campbell...      NaN      NaN   
9995                                                N/A      NaN      NaN   
9996  N/A    0 Orlando von EinsiedelN/A| N/A    Star...      NaN      NaN   
9997  N/A    0 Jovanka VuckovicN/A| N/A    1 Morgan ...      NaN      NaN   
9998  N/A    0 Jovanka VuckovicN/A| N/A    1 Morgan ...      NaN      NaN   

     Start Year Series  
0          2021     no  
1          2021    yes  
2          2010    yes  
3          2013    yes  
4          2021     no  
...         ...    ...  
9994       2021    yes  
9995       2021    yes  
9996       2022    yes  
9997       2021    yes  
9998       2021    yes  

[9999 rows x 8 columns]

In [30]:
df3['STARS'] = df3['STARS'].str.replace('N/A|',' ', regex=False)
df3.head(2)

MOVIES  \
0                        Blood Red Sky   
1  Masters of the Universe: Revelation   

                                      GENRE  RATING  \
0      Action, Horror, Thriller                 6.1   
1  Animation, Action, Adventure                 5.0   

                                               STARS   VOTES  RunTime  \
0  N/A    0 Peter Thorwarth  N/A    1 Peri Baumei...  21,062    121.0   
1  N/A            N/A    1 Chris Wood, N/ASarah M...  17,870     25.0   

  Start Year Series  
0       2021     no  
1       2021    yes

In [31]:
df3.STARS[9996]

'N/A    0 Orlando von Einsiedel  N/A    Star:N/APrince HarryN/A'

In [32]:
df3['STARS'] = df3['STARS'].str.replace('Star:N/A','1 ', regex=False)

In [33]:
df3.STARS[9996]

'N/A    0 Orlando von Einsiedel  N/A    1 Prince HarryN/A'

In [34]:
df3.loc[df3['STARS'] == 'N/A']

MOVIES  \
24    He-Man and the Masters of the Universe   
1027             Myth & Mogul: John DeLorean   
1380          Love is Blind: After the Altar   
1419              Avatar: The Last Airbender   
1454     Cocaine Cowboys: The Kings of Miami   
...                                      ...   
9972                             Sonic Prime   
9973                             Sonic Prime   
9974                             Sonic Prime   
9986                        The Last Kingdom   
9995                                  Arcane   

                                          GENRE  RATING STARS VOTES  RunTime  \
24     Animation, Action, Adventure                 NaN   N/A   NaN      NaN   
1027  Documentary, Biography, Crime                 7.0   N/A   213      NaN   
1380                     Reality-TV                 5.4   N/A    39     45.0   
1419     Action, Adventure, Fantasy                 NaN   N/A   NaN      NaN   
1454    Documentary, Crime, History                 NaN   N/A   NaN      NaN   
...                                         ...     ...   ...   ...      ...   
9972   Animation, Action, Adventure                 NaN   N/A   NaN      NaN   
9973   Animation, Action, Adventure                 NaN   N/A   NaN      NaN   
9974   Animation, Action, Adventure                 NaN   N/A   NaN      NaN   
9986         Action, Drama, History                 NaN   N/A   NaN      NaN   
9995   Animation, Action, Adventure                 NaN   N/A   NaN      NaN   

     Start Year Series  
24         2021    yes  
1027       2021    yes  
1380       2021    yes  
1419        NaN     no  
1454       2021     no  
...         ...    ...  
9972       2022    yes  
9973       2022    yes  
9974       2022    yes  
9986       2015    yes  
9995       2021    yes  

[456 rows x 8 columns]

In [35]:
df3.loc[df3['STARS'] == 'N/A', 'STARS'] = 'N/A 1 N/A'

In [36]:
df3

MOVIES  \
0                           Blood Red Sky   
1     Masters of the Universe: Revelation   
2                        The Walking Dead   
3                          Rick and Morty   
4                         Army of Thieves   
...                                   ...   
9994                       The Imperfects   
9995                               Arcane   
9996                    Heart of Invictus   
9997                       The Imperfects   
9998                       The Imperfects   

                                         GENRE  RATING  \
0         Action, Horror, Thriller                 6.1   
1     Animation, Action, Adventure                 5.0   
2          Drama, Horror, Thriller                 8.2   
3     Animation, Adventure, Comedy                 9.2   
4            Action, Crime, Horror                 NaN   
...                                        ...     ...   
9994     Adventure, Drama, Fantasy                 NaN   
9995  Animation, Action, Adventure                 NaN   
9996            Documentary, Sport                 NaN   
9997     Adventure, Drama, Fantasy                 NaN   
9998     Adventure, Drama, Fantasy                 NaN   

                                                  STARS    VOTES  RunTime  \
0     N/A    0 Peter Thorwarth  N/A    1 Peri Baumei...   21,062    121.0   
1     N/A            N/A    1 Chris Wood, N/ASarah M...   17,870     25.0   
2     N/A            N/A    1 Andrew Lincoln, N/ANor...  885,805     44.0   
3     N/A            N/A    1 Justin Roiland, N/AChr...  414,849     23.0   
4     N/A    0 Matthias Schweighöfer  N/A    1 Matth...      NaN      NaN   
...                                                 ...      ...      ...   
9994  N/A            N/A    1 Morgan Taylor Campbell...      NaN      NaN   
9995                                          N/A 1 N/A      NaN      NaN   
9996  N/A    0 Orlando von Einsiedel  N/A    1 Princ...      NaN      NaN   
9997  N/A    0 Jovanka Vuckovic  N/A    1 Morgan Tay...      NaN      NaN   
9998  N/A    0 Jovanka Vuckovic  N/A    1 Morgan Tay...      NaN      NaN   

     Start Year Series  
0          2021     no  
1          2021    yes  
2          2010    yes  
3          2013    yes  
4          2021     no  
...         ...    ...  
9994       2021    yes  
9995       2021    yes  
9996       2022    yes  
9997       2021    yes  
9998       2021    yes  

[9999 rows x 8 columns]

In [37]:
df3['STARS'].str.contains('1').sum()

9662

In [ ]:
#Current dilemma is that around 337 rows do not have '1' in them (as found from above)
#Need to figure out how to identify the index of those rows
#Then i want to check what i need to edit to change them to have ones
#This should allow me to then seperate the columns due to the '1'

In [38]:
df3[~df3["STARS"].str.contains("1")]

MOVIES  \
1367               El caso Wanninkhof-Carabantes   
1469                                   Pray Away   
1658  The Chronicles of Narnia: The Silver Chair   
2052              If Anything Happens I Love You   
2178                Untold: Malice at the Palace   
...                                          ...   
9988                            The Last Kingdom   
9989                                        1899   
9990                                        1899   
9991                                        1899   
9992                                        1899   

                                        GENRE  RATING  \
1367  Documentary, Crime, History                 6.5   
1469                  Documentary                 6.2   
1658           Adventure, Fantasy                 NaN   
2052      Animation, Short, Drama                 7.8   
2178           Documentary, Sport                 NaN   
...                                       ...     ...   
9988       Action, Drama, History                 NaN   
9989       Drama, History, Horror                 NaN   
9990       Drama, History, Horror                 NaN   
9991       Drama, History, Horror                 NaN   
9992       Drama, History, Horror                 NaN   

                                                  STARS   VOTES  RunTime  \
1367                            N/A    0 Tània BallóN/A   1,357     88.0   
1469                      N/A    0 Kristine StolakisN/A     159    101.0   
1658                           N/A    0 Joe JohnstonN/A     NaN      NaN   
2052  N/A    Directors:N/AMichael Govier, N/AWill Mc...  13,086     13.0   
2178                             N/A    0 Floyd RussN/A     NaN     68.0   
...                                                 ...     ...      ...   
9988                      N/A    0 Anthony PhilipsonN/A     NaN      NaN   
9989                          N/A    0 Baran bo OdarN/A     NaN      NaN   
9990                          N/A    0 Baran bo OdarN/A     NaN      NaN   
9991                          N/A    0 Baran bo OdarN/A     NaN      NaN   
9992                          N/A    0 Baran bo OdarN/A     NaN      NaN   

     Start Year Series  
1367       2021     no  
1469       2021     no  
1658        NaN     no  
2052       2020     no  
2178       2021     no  
...         ...    ...  
9988       2015    yes  
9989       2022    yes  
9990       2022    yes  
9991       2022    yes  
9992       2022    yes  

[337 rows x 8 columns]

Noticed that RunTime, Votes and Rating are NaN for some of these rows, Would it be better to remove the rows?
df4 = df3.dropna(subset=['RATING', 'VOTES', 'RunTime'])

In [44]:
#df4 = df3.dropna(subset=['RATING', 'VOTES', 'RunTime'], thresh=len(['RATING', 'VOTES', 'RunTime']))
df4 = df3.dropna(subset=['RATING', 'VOTES', 'RunTime'])
df4

MOVIES  \
0                           Blood Red Sky   
1     Masters of the Universe: Revelation   
2                        The Walking Dead   
3                          Rick and Morty   
5                             Outer Banks   
...                                   ...   
9959                               Sexify   
9960                               Sexify   
9961                               Sexify   
9962                               Sexify   
9963                          Castlevania   

                                         GENRE  RATING  \
0         Action, Horror, Thriller                 6.1   
1     Animation, Action, Adventure                 5.0   
2          Drama, Horror, Thriller                 8.2   
3     Animation, Adventure, Comedy                 9.2   
5             Action, Crime, Drama                 7.6   
...                                        ...     ...   
9959                 Comedy, Drama                 6.8   
9960                 Comedy, Drama                 7.1   
9961                 Comedy, Drama                 6.9   
9962                 Comedy, Drama                 7.1   
9963  Animation, Action, Adventure                 8.2   

                                                  STARS    VOTES  RunTime  \
0     N/A    0 Peter Thorwarth  N/A    1 Peri Baumei...   21,062    121.0   
1     N/A            N/A    1 Chris Wood, N/ASarah M...   17,870     25.0   
2     N/A            N/A    1 Andrew Lincoln, N/ANor...  885,805     44.0   
3     N/A            N/A    1 Justin Roiland, N/AChr...  414,849     23.0   
5     N/A            N/A    1 Chase Stokes, N/AMadel...   25,858     50.0   
...                                                 ...      ...      ...   
9959  N/A    0 Kalina Alabrudzinska  N/A    1 Aleksa...      108     38.0   
9960  N/A    0 Kalina Alabrudzinska  N/A    1 Aleksa...      103     37.0   
9961  N/A    0 Kalina Alabrudzinska  N/A    1 Aleksa...       96     40.0   
9962  N/A    0 Kalina Alabrudzinska  N/A    1 Aleksa...      100     47.0   
9963  N/A    0 Sam Deats  N/A    1 Richard Armitage,...      911     24.0   

     Start Year Series  
0          2021     no  
1          2021    yes  
2          2010    yes  
3          2013    yes  
5          2020    yes  
...         ...    ...  
9959       2021    yes  
9960       2021    yes  
9961       2021    yes  
9962       2021    yes  
9963       2017    yes  

[6786 rows x 8 columns]

In [ ]:
#This would remove 1/3 of the data, so should probably not drop it